test/AutoRest.TestServer.Tests/Mgmt/TestProjects/TestProjectTests.cs

```diff
@@ -363,6 +368,7 @@ public void ValidateSubscriptionExtensionsListResource()
                 {
                     var listMethodInfo = subscriptionExtension.GetMethod($"List{resourceName}", BindingFlags.Static | BindingFlags.Public);
                     Assert.NotNull(listMethodInfo);
+                    // This looks weird, some list functions have extra optional parameters (top, statusOnly, etc), therefore we cannot say we have 2 parameters for all list functions.
                     Assert.AreEqual(2, listMethodInfo.GetParameters().Length);
```

Comment:
As mentioned in above comments, this assertion does not make much sense, and this will be inherited by all the test cases, which creates unexpected test failures

Reply:
Also this